tensorflow_serving/batching/batching_session.h

```diff
@@ -99,6 +99,23 @@ struct BatchingSessionOptions {
   //
   // If left empty, no rounding/padding is performed.
   std::vector<int> allowed_batch_sizes;
+
+  // If set to true, padding is performed for tensors of the same name
+  // but with unequal dimensions, so that all tensors of the same name have
+  // equal dim sizes.
```

Comment:
Please explain what data is used for the padding, and what data types are supported.

For numerical types I believe you pad with zeros.

For non-numerical types (e.g. string; are there any others?) what do you do -- simply not supported? Or pad with empty strings?

For the padding we do for allowed_batch_sizes we use existing values (not zeros) as the padding, because we know that these are legal values (or if not, the processing will fail for reasons not having to do with the padding). Perhaps we should do the same thing for this padding? WDYT?